src/ringcentral-web-phone.js

```diff
@@ -200,6 +214,120 @@
+                    return;
+                }
+
+                var randomMinInterval = (this.configuration.reconnectionTimeout - 2 ) * 1000;
+                var randomMaxInterval = (this.configuration.reconnectionTimeout + 2 ) * 1000;
+
+                this.reconnectionAttempts += 1;
+                this.nextReconnectInterval = this.computeRandomTimeout(this.reconnectionAttempts, randomMinInterval, randomMaxInterval);                
+
+                if (this.reconnectionAttempts > this.configuration.maxReconnectionAttempts) {
+                    this.logger.warn('maximum reconnection attempts for WebSocket ' + this.server.ws_uri);
+                    this.logger.log('transport ' + this.server.ws_uri + ' failed | connection state set to \'error\'');
+                    this.server.isError = true;
+                    this.emit('transportError');
+                    this.server = this.getNextWsServer();
+                    this.reconnectionAttempts = 0;
+
+                    if (this.server !== this.configuration.wsServers[0]){
+                        this.scheduleSwithBackMainProxy();
+                    }
+
+                    this.connect();
+                } else {
+                    this.logger.log('trying to reconnect to WebSocket ' + this.server.ws_uri + ' (reconnection attempt ' + this.reconnectionAttempts + ')');
+                    this.reconnectTimer = SIP.Timers.setTimeout(function () {
+                        this.connect();
+                        this.reconnectTimer = null;
+                    }.bind(this), this.nextReconnectInterval);
+                    this.logger.warn('next reconnection attempt in:' + Math.round(this.nextReconnectInterval / 1000));
+                }                
+            };
+
+            transport.isSipErrorCode = function(message){
+                var statusLine = message.substring(0, message.indexOf('\r\n'));
+                var statusCode = statusLine.split(' ')[1];
+                return statusCode && this.sipErrorCodes && this.sipErrorCodes.length && this.sipErrorCodes.includes(statusCode);
+            };
+
+            transport.scheduleSwithBackMainProxy = function(attempts){
+               
+                var switchBackInterval = parseInt(this.switchBackInterval) ?  parseInt(this.switchBackInterval) * 1000 : null;
+              
+                if (switchBackInterval) {
+                    // Add random time to expand clients connections in time;
+                    switchBackInterval += this.computeRandomTimeout(1, 0, 15*60*1000);
+                    this.logger.warn('Try to switch back to main proxy after ' + Math.round(switchBackInterval/1000/60) + ' min');
+
+                    SIP.Timers.setTimeout(function(){
+                        this.configuration.wsServers[0].isError = false;
+                        //Notify client to trigger switching back to main proxy server;
+                        this.emit('switchBackProxy');
```

Comment:
How should the client handle this? Should the client disconnect the active wsserver and then reconnect to the primary wsserver ?